File: knowledge-gap-platform/src/main/java/com/knowledgegap/knowledge_gap_platform/service/QuestionService.java
package com.knowledgegap.knowledge_gap_platform.service;

import com.knowledgegap.knowledge_gap_platform.dto.assessment.QuestionResponse;

import java.util.List;

public interface QuestionService {
    List<QuestionResponse> getQuestionsByAssessment(Long assessmentId);

    QuestionResponse getQuestionById(Long questionId);
}

File: knowledge-gap-frontend/src/services/questionService.js
import api from "./api";

const BASE_URL = "api/questions";

const questionService = {

  getQuestionsByAssessment: (assessmentId) =>
    api.get(`${BASE_URL}/assessment/${assessmentId}`),

  getQuestionById: (questionId) =>
    api.get(`${BASE_URL}/${questionId}`)
};

export default questionService;